internal/control_plane/server/execution/execution.go

```diff
@@ -25,22 +23,19 @@ func NewExec(dbClient etcd.EtcdClient) Execution {
 	}
 }
 
-func (e *execution) CreateProc(ctx context.Context, proc *model.Proc) (string, error) {
-	
+func (e *execution) CreateProc(ctx context.Context, proc *protobuf.Proc) (string, error) {
+
 	result, err := e.client.PutValue(ctx, proc.Name, proc)
```

Comment:
You can return directly since it's the return type is the same.

```suggestion
	return e.client.PutValue(ctx, proc.Name, proc)
```

Reply:
Yeah. That was not intended to be there. We have now created a proto message construct and pass messages and errors accordingly